### PR TITLE
Multimedia test cleanup

### PR DESCRIPTION
**Pull request by benrudolph**
@snopoke reworking some of these tests to use the interface. also feel like the `interface.py` file is outliving its usefulness. also i realize this is just working on the tests in isolation, but i just need to read more of the code and understand it before i can feel comfortable correctly refactoring the data access points in the current form processing. will be trying to refactor soon. how do you feel about this sort of structure

```
form_processor/interfaces/xform.py
form_processor/interfaces/case.py
form_processor/interfaces/sync_log.py
form_processor/interfaces/__init__.py   # inherits the 3 different interfaces
```

only saying this cause i find myself writing `get_case_from_generic`, `get_form_from_generic`, and `get_sync_log_from_generic`, so it seems like they should be split out.

buddy: @esoergel 

probably best read commit-by-commit

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py [hunk 1]
```diff
@@ -82,39 +80,40 @@ def _do_submit(self, xml_data, dict_attachments, sync_token=None, date=None):
...
         [case] = cases
         self.assertEqual(case.case_id, TEST_CASE_ID)
 
+        return response, form, cases
+
     def _submit_and_verify(self, doc_id, xml_data, dict_attachments,
                            sync_token=None, date=None):
-        self._do_submit(xml_data, dict_attachments, sync_token, date=date)
-
-        time.sleep(2)
```
Comment (benrudolph) [inline corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py]: this was fun

**Comment by benrudolph** [comment]
... no idea why tests aren't being run

File: corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py (hunk 1 above)
Comment (czue) [inline corehq/ex-submodules/casexml/apps/case/tests/test_multimedia.py]: lol

**Comment by benrudolph** [comment]
:white_check_mark:

**Comment by esoergel** [comment]
Looks good to me, but I'll leave for @snopoke, who it sounds like has more context on this.

**Pull request merged**